fintech-examples/fintech-server/src/test/java/de/adorsys/opba/fintech/server/FinTechServerTests.java

```diff
@@ -0,0 +1,261 @@
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@EnableFinTechImplConfig
+@Slf4j
+class FinTechServerTests {
+    private static final String FIN_TECH_AUTH_URL = "/v1/login";
+    private static final String FIN_TECH_BANK_SEARCH_URL = "/v1/search/bankSearch";
+    private static final String FIN_TECH_BANK_PROFILE_URL = "/v1/search/bankProfile";
+
+    private static final String BANK_SEARCH_RESPONSE_PREFIX = "TPP_BankSearchResponse";
+    private static final String BANK_PROFILE_RESPONSE_PREFIX = "TPP_BankProfileResponse";
+    private static final String POSTFIX = ".json";
+
+    private static final Gson gson = new Gson();
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Autowired
+    protected MockMvc mvc;
+
+    @Test
+    @SneakyThrows
+    public void loginPostOk() {
+        auth("peter", "1234");
+    }
+
+    /**
+     *
+     * @param username
+     * @param password
+     * @return XSRF Token
+     */
+    @SneakyThrows
+    private String auth(String username, String password) {
+        LoginBody loginBody = new LoginBody(username, password);
+        MvcResult result = this.mvc
+                .perform(MockMvcRequestBuilders.post(FIN_TECH_AUTH_URL)
+                        .header("X-Request-ID", UUID.randomUUID().toString())
+                        .content(gson.toJson(loginBody))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.header().exists("X-XSRF-TOKEN"))
+                .andReturn();
+        return result.getResponse().getHeader("X-XSRF-TOKEN");
+    }
+
+    @Test
+    @SneakyThrows
+    public void loginPostUnAuthorized() {
+        final LoginBody loginBody = new LoginBody("peter", "12345");
+        this.mvc
+                .perform(MockMvcRequestBuilders.post(FIN_TECH_AUTH_URL)
+                        .header("X-Request-ID", UUID.randomUUID().toString())
+                        .content(gson.toJson(loginBody))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isUnauthorized())
+                .andExpect(MockMvcResultMatchers.header().doesNotExist("X-XSRF-TOKEN"));
+    }
+
+    @Mock
+    TppBankSearchApi mockedTppBankSearchApi;
```

Comment:
You can use MockBean annotation on this, so it will be autowired instead of setTppBankSearchApi

Reply:
Thanks, done